src/html/classic/ng/src/web/pages/extras/trashactions.js

```diff
@@ -0,0 +1,181 @@
+/* Greenbone Security Assistant
+ *
+ * Authors:
+ * Steffen Waterkamp <steffen.waterkamp@greenbone.net>
+ *
+ * Copyright:
+ * Copyright (C) 2017 Greenbone Networks GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import React from 'react';
+import _ from 'gmp/locale.js';
+
+import IconDivider from '../../components/layout/icondivider.js';
+import RestoreIcon from '../../components/icon/restoreicon.js';
+import TrashDeleteIcon from '../../components/icon/trashdeleteicon.js';
+
+import TableData from '../../components/table/data.js';
+
+import {is_defined} from 'gmp/utils.js';
+import PropTypes from '../../utils/proptypes.js';
+
+const TrashActions = ({
+    entity,
+    onEntityDelete,
+    onEntityRestore,
+  }) => {
+
+    const check_by_type = {
```

Comment:
Please move the check_by_type object out of TashActions and pass the entity as a parameter to the functions. They object is generated on each render and should only be generated once because it is static.

Reply:
Please use arrow functions instead.
```javascript
agent: entity => {
  return {restorable, deletable: entity.isInUse()};
},
```

Reply:
Moved check_by_type out of TrashActions, unscrambled the logics for "deletable", and rewrote to use arrow functions.